### DISCUSSION TITLE
Ways to shorten the computational time for the initial MD step and other ideas

### DISCUSSION THREAD

**Discussion by tobigithub** [qcxms/QCxMS#31] [Ideas]
Hi,
the first initial MD step, before the production run, takes considerable time for CID. While this may not play a role for runs on compute clusters, it is a considerable slow-down on single CPUs. Approach (1) is probably worthwhile to do, approaches (2) and (3) probably ideas that should be discussed.

**(1)** So what is the best way to shorten the computational times? Select MNDO99 or MOPAC for the first QCxMS step and then switch back to GFN-XTB for the trajectory generation (second qcxms call) and also the final "qcxms -prod", the production run? Also I wonder if its possible to sample the ground state modes  with faster methods such as forcefield like the [GFN-FF](https://xtb-docs.readthedocs.io/en/latest/gfnff.html#gfnff). So conformer generation with GFN-FF and then temperature calibration with few higher level MD steps if needed. 

**(2)** Also I wonder what the results would be to to all the MD steps in GFN-FF and just do the energy and force calculations in GFN-XTB. Supposedly [GFN-FF is a dissociative force field](https://onlinelibrary.wiley.com/doi/full/10.1002/anie.202004239). 

**(3)** Another idea is basically a hybrid QM/MM approach or something similar to [ONIUM calculations](https://en.wikipedia.org/wiki/ONIOM) in Gaussian or NWCHEM. Also, similar processes were already implemented into QCEIMS, allowing the MD steps performed in semiempiricals and the IP/IE calculations with DFT.

But there could be also a more radical approach, for example in CID, when the ion accelerates towards the neutral collision gas, we do not need full DFT treatment of the MD process I think. There are exceptions, for example when rearrangements occur before the collision, or when charge transfer reactions take place. However, before the collision the MD steps can be switched to a very cheap method, like GFN-FF and once the molecules are near each other and start interacting, DFT-MD can be switched back on.  Of course in the next step, many interesting reactions occur in the mean free path, so switching back to lower level of theory methods probably won't be beneficial here. But who knows, one could do an mixed approach, running 10 MD steps with a lower level of theory and then switching back to higher level of theory for one step, repeatedly. 

Another idea is of course also the inclusion of machine learning force fields like [sGDML](https://www.sciencedirect.com/science/article/pii/S0010465519300591) for certain parts of the computation.  Anyway, just some ideas, hybrid QM/MM and ONIUM have worked well for large molecules, so QCxMS could probably also see benefits, implementations are of course much more complicated, so that's not something to code over the weekend. 
Tobias

**Comment by JayTheDog** [comment]
The groundstate MD simulation time depends on the number of trajectories and can be reduced by using less trajectories. It can be worthwhile to check if a low number of trajectories will provide qualitatively the same result. The more rigid the molecule, the lower number of trajectories can be used. 
Running MNDO or MOPAC is not what is intended, because the program than no longer remains independent from third party software. You can easily use the workflow you have described by yourself or write a script that sets the QC method by itself. However, I'm not sure why using anything other than GFN-xTB is faster?
The equilibration is important for calculating MDs, but I agree that this first MD might be shortened. 
Using GFN-FF was discussed as an alternative while developing it, but would unfortunately not sufficiently describe the correct fragmentation procedures. 
Switching between methods can be beneficial, the first MD steps before collision can of course be calculated cheaper, but as you mentioned, as soon as the collision occurs, we need correct QC methods. So this would only be a very small improvement on the overall timings. Taking a hybrid method might be an interesting  approach and I think worthwhile to check. A problem can be that fragmentation reactions can be computed somewhat wrong, but this has to be checked before ruling it out. An idea already on the to-do list is to use DFT for calculating IPs in very close cases, which might reduce the amount of wrongfully determined artifacts.
